rust/arrow/src/ipc/reader.rs

```diff
@@ -793,7 +793,21 @@ impl<R: Read> StreamReader<R> {
         }
         // determine metadata length
         let mut meta_size: [u8; 4] = [0; 4];
-        self.reader.read_exact(&mut meta_size)?;
+
+        match self.reader.read_exact(&mut meta_size) {
+            Ok(()) => (),
+            Err(e) => {
+                if e.kind() == std::io::ErrorKind::UnexpectedEof {
+                    // Handle EOF without the "0xFFFFFFFF 0x00000000"
+                    // valid according to:
+                    // https://arrow.apache.org/docs/format/Columnar.html#ipc-streaming-format
+                    self.finished = true;
+                    return Ok(None);
+                }
+                return Err(ArrowError::from(e));
```

Comment:
nit: could you remove the `return` statements?

perhaps:

```rust
if e.kind() == std::io::ErrorKind::UnexpectedEof {
  ...
  Ok(None)
} else {
  Err(ArrowError::from(e))
}
```

Another option:

```rust
Err(e) if e.kind() == std::io::ErrorKind::UnexpectedEof => {
  ...
  Ok(None)
}
Err(e) => Err(ArrowError::from(e))
```

Reply:
I don't think that i could remove return  , As it's not the last statement of the function .  

for example i tried this : 
`if let Err(e) = self.reader.read_exact(&mut meta_size) {
            if e.kind() == std::io::ErrorKind::UnexpectedEof {
                // Handle EOF without the "0xFFFFFFFF 0x00000000"
                // valid according to:
                // https://arrow.apache.org/docs/format/Columnar.html#ipc-streaming-format
                self.finished = true;
                Ok(None)
            } else {
                Err(ArrowError::from(e))
            }
        } `

but the complier complain : 
`expected `()`, found enum `std::result::Result``

Reply:
I see. This is returning from the whole function not the match .. ok, never mind then and thanks for looking at it.